libcontainer/rootfs_linux.go

```diff
@@ -572,6 +572,13 @@ func pivotRoot(rootfs, pivotBaseDir string) error {
 	if err != nil {
 		return fmt.Errorf("can't create pivot_root dir %s, error %v", pivotDir, err)
 	}
+	defer func() error {
+		err := os.Remove(pivotDir)
+		if err != nil {
+			return err
+		}
+		return nil
+	}()
```

Comment:
I don't think this is right, defer can alter the returned parameters, but you can't return a new set.
Maybe you should change the function to use named result parameters.

Also, we should not alway call `os.Remove(pivotDir)` in defer because in some cases, `pivotDir` is not even defined.